Readme for Webdav service (#4912)
owncloud/ocis: services/webdav/README.md

# Webdav Service

> The webdav service (like the [ocdav](https://github.com/owncloud/ocis/tree/master/services/ocdav) service) provides an http API following the webdav protocol. 
> 
> ## Functionality
> 
> It receives http calls from clients and issues grpc calls to other services to execute the requests.
> After the other services have finished it will render their responses in `xml` and send them back to the client.

Reviewer: ```suggestion
The webdav service, like the [ocdav](https://github.com/owncloud/ocis/tree/master/services/ocdav) service, provides a HTTP API following the webdav protocol. It receives HTTP calls from requestors like clients and issues gRPC calls to other services executing these requests. After the called service has finished the request, the webdav service will render their responses in `xml` and sends them back to the requestor.
```

## Endpoints Overview

> Currently webdav service handles request for two functionalities: `Thumbnails` and `Search`

Reviewer: ```suggestion
Currently, the webdav service handles request for two functionalities, which are `Thumbnails` and `Search`.
```

### Thumbnails

> Webdav service provides various `GET` endpoints to get the thumbnails of a file in authenticated and unauthenticated contexts. It also provides thumbnails for spaces on different endpoints. 
> 
> See [Thumbnails](https://github.com/owncloud/ocis/tree/master/services/thumbnails) service for more information about thumbnails

Reviewer: ```suggestion
The webdav service provides various `GET` endpoints to get the thumbnails of a file in authenticated and unauthenticated contexts. It also provides thumbnails for spaces on different endpoints. 

See the [thumbnail](https://github.com/owncloud/ocis/tree/master/services/thumbnails) service for more information about thumbnails.
```

### Search

> Webdav service provides access to the search functionality. It offers multiple `REPORT` endpoints for getting search results. 
> 
> See [Search](https://github.com/owncloud/ocis/tree/master/services/search) service for more details about search functionality. 

Reviewer: ```suggestion
The webdav service provides access to the search functionality. It offers multiple `REPORT` endpoints for getting search results. 

See the [search](https://github.com/owncloud/ocis/tree/master/services/search) service for more details about search functionality. 
```

## Scalability

> Webdav service does not persist any data and does not cache any information. Therefore multiple instances of this service can be spawned in a bigger deployment (e.g. kubernetes) without any extra configuration.

Reviewer: ```suggestion
The webdav service does not persist any data and does not cache any information. Therefore multiple instances of this service can be spawned in a bigger deployment like when using container orchestration with Kubernetes, without any extra configuration.
```
